frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+      tabCell.setAttribute('id', 'dataType' + i + k);
+      if (sp_table.ColDefs[currentColumnSp].IsArray == true)
+        tabCell.innerHTML = 'ARRAY(' + sp_table.ColDefs[currentColumnSp].T.Name + ')';
+      else {
+        tabCell.innerHTML = sp_table.ColDefs[currentColumnSp].T.Name;
+      }
+    }
+  }
+}
+
+/**
+ * Function to create constraint column for spanner table
+ *
+ * @param {number} i table index
+ * @param {number} k column index in table
+ * @return {null}
+ */
+const constraintIterator = (i, k) => {
+  for (var l = 0; l < 2; l++) {
+    tabCell = tr.insertCell(-1);
+    tabCell.className = "acc-table-td";
+    if (l % 2 == 0) {
+      count_src[i][k] = 0;
+      srcPlaceholder[z] = count_src[i][k];
+      if (src_table.ColDefs[currentColumnSrc].NotNull != undefined) {
+        if (src_table.ColDefs[currentColumnSrc].NotNull == true) {
+          count_src[i][k] = count_src[i][k] + 1;
+          srcPlaceholder[z] = count_src[i][k];
+          notNullFound = "<option disabled class='active'>Not Null</option>";
+        }
+        else {
+          notNullFound = "<option disabled>Not Null</option>";
+        }
+      }
+      else {
+        notNullFound = '';
+      }
+
+      constraintId = 'srcConstraint' + z;
+      srcConstraintHtml = "<select id=" + constraintId + " multiple size='1' class='form-control spanner-input tableSelect'>"
+        + notNullFound
+        + "</select>";
+      tabCell.innerHTML = srcConstraintHtml;
+      z++;
+    }
+    else {
+      count_sp[i][k] = 0;
+      spPlaceholder[i][k] = count_sp[i][k];
+      // checking not null consraint
+      if (sp_table.ColDefs[currentColumnSp].NotNull != undefined) {
+        if (sp_table.ColDefs[currentColumnSp].NotNull == true) {
+          count_sp[i][k] = count_sp[i][k] + 1
+          spPlaceholder[i][k] = count_sp[i][k];
+          notNullFound = "<option disabled class='active'>Not Null</option>";
+          notNullFoundFlag[i][k] = true;
+        }
+        else {
+          notNullFound = "<option disabled>Not Null</option>";
+          notNullFoundFlag[i][k] = false;
+        }
+      }
+      else {
+        notNullFound = "<option disabled>Not Null</option>";
+        notNullFoundFlag[i][k] = false;
+      }
+      constraintId = 'spConstraint' + i + k;
+      spConstraintHtml = "<select id=" + constraintId + " multiple size='1' class='form-control spanner-input tableSelect'>"
+        + notNullFound
+        + "</select>";
+      tabCell.innerHTML = spConstraintHtml;
+      tabCell.setAttribute('class', 'sp-column acc-table-td spannerTabCell' + i + k);
+      constraintTabCell[i][k] = tabCell.innerHTML;
+    }
+  }
+}
+
+/**
+ * Function to create foreign key tab for each table
+ *
+ * @param {number} index table index
+ * @param {Array} foreignKeys foreign keys array for each table
+ * @return {null}
+ */
+const foreignKeyHandler = (index, foreignKeys) => {
+
+}
```

Comment:
Can we remove this function now?

Reply:
No this function is required. It is created and pushed now

Reply:
oh i see.